components/org.apache.stratos.mock.iaas.client/src/main/java/org/apache/stratos/mock/iaas/client/MockIaasApiClient.java

```diff
@@ -48,6 +49,20 @@ public MockIaasApiClient(String endpoint) {
         this.endpoint = endpoint;
     }
 
+    public boolean isMockIaaSReady() {
+        try {
+            URI uri = new URIBuilder(endpoint + INIT_CONTEXT).build();
+            HttpResponse response = restClient.doGet(uri);
+            if (response != null) {
+                return response.getStatusCode() == 200;
+            }
+            throw new RuntimeException("An unknown error occurred");
+        } catch (Exception e) {
+            String message = "Could not check whether mock-iaas is active";
+            throw new RuntimeException(message, e);
```

Comment:
RuntimeException is used all over this class. We will fix this in a separate commit.